### ISSUE TITLE
Hot module reloading does not work in 8.0.x

### ISSUE THREAD

**Issue by malyzeli** [styleguidist/react-styleguidist#1206]
**Current behavior**

Hot module reloading does not work in default configuration, tested in minimal project environment.

Seems it is caused by some Webpack caching bug. HMR works again when `cache: false` option is added to `webpackConfig` section. I'm not sure what are the consequences of that. 

**To reproduce**

I created minimal example project so anybody can look further into it. 

Project link: https://github.com/malyzeli/styleguidist-issue-1188-example

See related #1188 for more info.

**Expected behavior**

If I remember correctly, HMR used to work without having to explicitly configure styleguidist.

**Comment by sapegin** [comment]
Duplicate of #1188. Feel free to send a pull request with a fix 👾

**Comment by malyzeli** [comment]
@sapegin Honestly, I would argue that **this** is the real issue, while #1188 should be labeled as duplicate. My reasoning is that description of #1188 suggests that the problem affects Create React App projects only, while the problem is much more essential because **hot module reloading does not work by default** as I reported here (while the official documentation says Styleguidist provides live development environment).

I believe **more developers would look into that if stated explicitly that HMR is broken** in general, because this affects everybody, not just CRA users. Seems important to me.

I would also like to help solving the issue, but my knowledge of Webpack and Styleguidist internals is really minimal, I'm not even sure what really changes with disabling caching and what impact it may have on Styleguidist working. I did what I can - created minimal project, a playground for somebody who really knows what is he doing. :-)
